src/parser.rs

```diff
@@ -1318,6 +1460,19 @@ impl<'a> Parser<'a> {
     }
 }
 
+// Trim path start forward slashes when no authority is present
+// https://github.com/whatwg/url/issues/232
```

Comment:
I'm pretty sure I have some code somewhere that fixes the same tests as the ones in that WHATWG issue with less shenanigans, I'll try to find it again.

Reply:
There was indeed no need for an if else ^^'